pyro/infer/mcmc/hmc.py

```diff
@@ -304,12 +306,13 @@ def sample(self, params):
             # apply Metropolis correction.
             energy_proposal = self._kinetic_energy(r_new) + potential_energy_new
         delta_energy = energy_proposal - energy_current
-        # Set accept prob to 0.0 if delta_energy is `NaN` which may be
-        # the case for a diverging trajectory when using a large step size.
-        if torch_isnan(delta_energy):
-            accept_prob = scalar_like(delta_energy, 0.)
-        else:
-            accept_prob = (-delta_energy).exp().clamp(max=1.)
+        # handle the NaN case which may be the case for a diverging trajectory
+        # when using a large step size.
+        delta_energy = scalar_like(delta_energy, float("inf")) if torch_isnan(delta_energy) else delta_energy
+        if delta_energy > self._max_sliced_energy and self._t >= self._warmup_steps:
```

Comment:
Just want to confirm - the 1000 factor is being applied in log space, so the actual difference in energy is much higher?

Reply:
I guess you mean actual difference is accept prob? It's pretty small `exp(-1000) ~ 0`.

Reply:
Sorry, I was mistaken. The energy isn't in the log space - this is fine!